packages/next-auth/src/react/index.tsx

```diff
@@ -456,6 +462,21 @@ export function SessionProvider(props: SessionProviderProps) {
         : session
         ? "authenticated"
         : "unauthenticated",
+      async update(data) {
+        setLoading(true)
+        const session = await fetchData<Session>(
+          "session",
+          __NEXTAUTH,
+          logger,
+          { req: { body: { csrfToken: await getCsrfToken(), data } } }
+        )
+        setLoading(false)
```

Comment:
should call `setLoading(false)` in a `try` `finally` block in case the fetch fails

Reply:
`fetchData` already handles fetch related issues.

https://github.com/nextauthjs/next-auth/blob/cd91c03211f23fd8dec270ae911de9facb11b704/packages/next-auth/src/client/_utils.ts#L54C11-L55